### PR TITLE
Upgrade npm dependencies

### LINKED ISSUE #14450: npm audit shows warnings (run npm audit fix)
there's some packages with npm-detected security issues:

```
found 31 vulnerabilities (5 low, 19 moderate, 6 high, 1 critical) in 15716 scanned packages
  run `npm audit fix` to fix 6 of them.
  12 vulnerabilities require semver-major dependency updates.
  13 vulnerabilities require manual review. See the full report for details.
```

we should get the low-hanging fruit by running npm audit fix at the least

### PR DESCRIPTION
**Pull request by Slava**
Fixes #14450

Reduces the number of reported vulnerabilities from 28 down to 9. 6 are pretty innocent in the context of Brave (DoS by specific regexes in util functions, etc). But 3 would be fixed by upgrading `mocha` to 5.x. Unfortunately upgrading mocha breaks the unit tests and I couldn't figure out why.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by Slava** [comment]
@diracdeltas

**Comment by diracdeltas** [comment]
cool, let's wait a bit to see how travis does with this change. also please resolve the package-lock merge conflict

**Comment by bsclifton** [comment]
I have another patch coming which affects package-lock.json - so you might hold off on the merge conflict for a few mins

**Comment by Slava** [comment]
Sure, let's wait for @bsclifton's patch first. Also I might discover more problems with this upgrade in the meantime

**Comment by bsclifton** [comment]
@Slava finished! You're good to rebase now, sir 😄 👍

**Comment by Slava** [comment]
Mmmm heads up, after this upgrade `npm audit` command no longer works (gets a 400: `400 Bad Request - POST https://registry.npmjs.org/-/npm/v1/security/audits`) and every consecutive `rm -rf node_modules && npm install` changes the `package-lock.json` file beyond my understanding.

File: package.json [hunk 1]
```diff
@@ -201,7 +201,7 @@
     "style-loader": "~0.19.0",
     "uglifyjs-webpack-plugin": "^1.0.0",
     "uuid": "^3.0.1",
-    "webdriverio": "4.7.1",
+    "webdriverio": "^4.13.0",
```
Comment (diracdeltas) [inline package.json]: @NejcZdovc according to git blame, you set this to 4.7.1 for https://github.com/brave/browser-laptop/issues/10072. is that still a problem?

Comment (diracdeltas) [reply package.json]: looks like this is causing a bunch of deprecationWarnings but not errors (that weren't already there). https://travis-ci.org/brave/browser-laptop/jobs/394314325#L6366

can we set deprecationWarning false in this PR?

File: package.json [hunk 2]
```diff
@@ -168,19 +168,19 @@
     "flow-bin": "^0.53.1",
     "font-awesome-webpack": "~0.0.4",
     "git-rev-sync": "^1.8.0",
-    "gulp": "^3.9.0",
+    "gulp": "^4.0.0",
```
Comment (NejcZdovc) [inline package.json]: We can remove gulp. I didn't find usage anywhere

cc @bsclifton

Comment (Slava) [reply package.json]: what about other tools like Flow?

Comment (diracdeltas) [reply package.json]: @Slava i started setting up flow but it never really got used. i  don't think anything runs flow check anymore

Comment (bsclifton) [reply package.json]: yup- we can remove gulp. We do use it elsewhere (ex: when packaging our version of PDF.js), but not in browser-laptop

Comment (bsclifton) [reply package.json]: ++ on removing flow

Comment (Slava) [reply package.json]: the subsequent commit removes both (as well as references to flow plugins)

**Comment by bsclifton** [comment]
If we'd like this, we'll want to rebase the change against master (PR is basically against what is now the `0.25.x-legacy` branch)

File: package.json [hunk 3]
```diff
@@ -165,22 +163,19 @@
...
     "jsdom-global": "~3.0.2",
     "json-loader": "^0.5.4",
     "jsonfile": "^2.2.3",
-    "less": "^2.5.3",
+    "less": "^3.0.4",
     "less-loader": "^2.2.1",
     "lolex": "~1.3.2",
     "mkdirp": "^0.5.1",
-    "mocha": "^5.2.0",
+    "mocha": "^2.5.3",
```
Comment (diracdeltas) [inline package.json]: this regresses https://github.com/brave/browser-laptop/pull/15165 which causes the `1 critical vulnerability`

**Comment by bsclifton** [comment]
@diracdeltas understandable 😄 thanks for checking it out

**Pull request closed without merging**